public/app/plugins/panel/graph/graph.ts

```diff
@@ -428,7 +452,9 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
         let defaultTicks = panelWidth / 50;
 
         if (data.length && bucketSize) {
-          ticks = _.map(data[0].data, point => point[0]);
+          ticks = _.chain(data).map(d => {
+            return _.map(d.data, point => point[0]);
```

Comment:
that test was not very realistic. In a more realistic test (20 series with 2000 data points) the for loop was 59 times faster:
https://jsperf.com/js-map-for-uniq2/1